server/src/instant/jdbc/wal.clj

```diff
@@ -159,16 +202,45 @@
 
    We do some book-keeping for the replication stream, by recording the LSN
    for the last record that was pushed to `to`."
-  [stream to shutdown?]
-  (while (not @shutdown?)
-    (let [buffer (.readPending stream)]
+  [stream to close-signal-chan]
+  (loop []
+    (let [buffer (.read stream)]
```

Comment:
Some context on why this was `.readPending` before: 

https://stackoverflow.com/questions/76074375/safely-stopping-pgreplicationstream 

I wasn't able to 'interrupt' the thread when it was blocked on `.read`. I think `kick-wal` is a great goaround for this though! 

I remember searching all over github for good examples of PGReplicationStream. Glad we get to contribute this back into the collective conciousness.

Reply:
Interesting--I was always able to get `.close` to eventually complete, it just took a while (~20 seconds max) when there was no activity on the wal. Let me experiment a little more, though.

Reply:
`.close` always eventually completed for me, but the max was a little over a minute. 

If we set up `component` or something similar and we're doing a ton of restarting in development, I'll set up `kick-wal` to run whenever we restart. I updated `kick-wal` to just do `pg_notify`, which seems to work and is less dumb.